homeassistant/components/cloud/__init__.py

```diff
@@ -377,8 +380,10 @@ async def remote_prefs_updated(prefs: CloudPreferences) -> None:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up a config entry."""
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
-    stt_platform_loaded: asyncio.Event = hass.data[DATA_PLATFORMS_SETUP][Platform.STT]
-    stt_platform_loaded.set()
+    stt_tts_entities_added: asyncio.Event = hass.data[DATA_PLATFORMS_SETUP][
+        "stt_tts_entities_added"
+    ]
+    stt_tts_entities_added.set()
```

Comment:
What does setting this `Event` mean? `async_setup_entry` of the stt and tts platforms do not wait for the entities to be created; they just call `async_add_entities` which spawns a task.

If this Event is not waiting for the entities to be added, it should be renamed, if it is waiting I don't understand how it works.

Reply:
They do wait. It's part of the entity platform helper.